clang/include/clang/Basic/DiagnosticSemaKinds.td

```diff
@@ -11089,8 +11089,8 @@ def err_first_argument_to_cwsc_pdtor_call : Error<
 def err_second_argument_to_cwsc_not_pointer : Error<
   "second argument to __builtin_call_with_static_chain must be of pointer type">;
 
-def err_vector_incorrect_num_elements : Error<
-  "%select{too many|too few}0 elements in vector %select{initialization|operand}3 (expected %1 elements, have %2)">;
+def err_tensor_incorrect_num_elements : Error<
```

Comment:
Calling this tensor really feels wrong. You've updated the message to support vectors and matrices, but not tensors so the naming really feels way out of place.